From: Usama Arif <usamaarif642@gmail.com>
Subject: mm-free-zapped-tail-pages-when-splitting-isolated-thp-fix
Date: Thu, 15 Aug 2024 20:16:56 +0100

fix bug going from v1 page version to the folio version of the patch in v3

Signed-off-by: Usama Arif <usamaarif642@gmail.com>
Reported-by: Kairui Song <ryncsn@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/huge_memory.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/huge_memory.c~mm-free-zapped-tail-pages-when-splitting-isolated-thp-fix
+++ a/mm/huge_memory.c
@@ -3152,7 +3152,7 @@ static void __split_huge_page(struct pag
 
 			folio_clear_active(new_folio);
 			folio_clear_unevictable(new_folio);
-			if (!folio_batch_add(&free_folios, folio)) {
+			if (!folio_batch_add(&free_folios, new_folio)) {
 				mem_cgroup_uncharge_folios(&free_folios);
 				free_unref_folios(&free_folios);
 			}
_
